### PR TITLE
Add documentation about how to run the tests

### PR DESCRIPTION
**Pull request by damien-carcel**
## Description

This PR adds a new documentation in our `internal_doc` directory, listing all the tests we are currently using in the PIM, and more important, how to run them.

## Definition Of Done

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | -
| Added legacy Behats               | -
| Added acceptance tests            | -
| Added integration tests           | -
| Changelog updated                 | -
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | -
| Migration script                  | -
| Tech Doc                          | OK

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: internal_doc/RUNNING_THE_TESTS.md [hunk 1]
```diff
@@ -0,0 +1,303 @@
+# Running the tests in Akeneo PIM
+
+## Introduction
+
+The code source of Akeneo PIM is extensively tested, with static analysis and automated tests, both on the back-end and the front-end.
```
Comment (tamarasaurus) [inline internal_doc/RUNNING_THE_TESTS.md]: source code

File: internal_doc/RUNNING_THE_TESTS.md [hunk 2]
```diff
@@ -0,0 +1,303 @@
...
+This document presents a list of all the tools that are used to test Akeneo PIM, what kind of testing they are used for, and how to run them.
+
+All those tools are currently run on the Akeneo CI. For each tool, examples are provided, explaining how to run all the tests at once, only one, or a specific set.
```
Comment (tamarasaurus) [inline internal_doc/RUNNING_THE_TESTS.md]: you can remove the comma after 'examples are provided'

File: internal_doc/RUNNING_THE_TESTS.md [hunk 3]
```diff
@@ -0,0 +1,303 @@
...
+### PHP Coupling Detector
+
+Official website: https://github.com/akeneo/php-coupling-detector
+
+This tool, maintained by Akeneo, detects the coupling issues in the code base, according to the coupling rules defined in the project.
+By default, the rules are defined in a `.php_cd` file, placed at the root of the project.
+
+In Akeneo PIM, there is one file [at the root of the project](https://github.com/akeneo/pim-community-dev/blob/master/.php_cd.php),
+but also one per bounded context (read the [architecture documentation](https://github.com/akeneo/pim-community-dev/blob/master/internal_doc/ARCHITECTURE.md) to learn more about bounded contexts).
+The one at the root of the project will eventually disappear.
```
Comment (tamarasaurus) [inline internal_doc/RUNNING_THE_TESTS.md]: what do you mean by that ?

File: internal_doc/RUNNING_THE_TESTS.md [hunk 4]
```diff
@@ -0,0 +1,303 @@
...
+If you want to only run the rules on a specific file:
+```bash
+$ vendor/bin/php-coupling-detector detect --config-file=src/Akeneo/UserManagement/.php_cd.php src/Akeneo/UserManagement/Bundle/Controller/UserController.php
+```
+
+### PHP Coding Standards Fixer (php-cs-fixer)
+
+Official website: https://github.com/FriendsOfPHP/PHP-CS-Fixer
+
+This tool can both detect and fix issues in your code, according to the standard you defined. Akeneo PIM follows the PSR-2 standard, plus a few more rules.
```
Comment (tamarasaurus) [inline internal_doc/RUNNING_THE_TESTS.md]: you have

File: internal_doc/RUNNING_THE_TESTS.md [hunk 5]
```diff
@@ -0,0 +1,303 @@
...
+If you want to **fix** all the violations, remove the `--dry-run` option:
+```bash
+$ vendor/bin/php-cs-fixer fix --diff --config=.php_cs.php
+```
+
+### phpspec
+
+Official website: https://www.phpspec.net/en/stable/
+
+`phpspec` is used to unit testing Akeneo PIM. Test suites are defined in the file [phpspec.yml.dist](https://github.com/akeneo/pim-community-dev/blob/master/phpspec.yml.dist).
```
Comment (tamarasaurus) [inline internal_doc/RUNNING_THE_TESTS.md]: used to unit test

File: internal_doc/RUNNING_THE_TESTS.md [hunk 6]
```diff
@@ -0,0 +1,303 @@
...
+If you want to run a specific test, add the line where the test start:
+```bash
+$ vendor/bin/phpspec run src/Akeneo/UserManagement/Component/spec/Model/GroupSpec.php:42
+```
+
+### PHPUnit
+
+Official website: https://phpunit.de/
+
+In Akeneo PIM, `PHPUnit` is used for integration tests, and a few remaining, legacy unit tests that were not migrated to `phpspec`.
```
Comment (tamarasaurus) [inline internal_doc/RUNNING_THE_TESTS.md]: remove comma after 'a few remaining'

File: internal_doc/RUNNING_THE_TESTS.md [hunk 7]
```diff
@@ -0,0 +1,303 @@
...
+```
+
+### Jest + Enzyme
+
+Official websites:
+- Jest: https://jestjs.io/
+- Enzyme: https://github.com/airbnb/enzyme
+
+`Jest` is a JavaScript testing platform. In Akeneo PIM, it is used for both unit and integration front-end tests.
+`Enzyme` is used to help testing some specific part of the front-end code written with React.
```
Comment (tamarasaurus) [inline internal_doc/RUNNING_THE_TESTS.md]: to help test

File: internal_doc/RUNNING_THE_TESTS.md [hunk 8]
```diff
@@ -0,0 +1,303 @@
...
+```bash
+$ yarn run acceptance tests/features/volume-monitoring
+```
+
+If you want to run the acceptance tests from a specific file:
+```bash
+$ yarn run acceptance tests/features/volume-monitoring/monitor_attribute_per_family_volume.feature
+```
+
+If you want to run a specific acceptance test, add the line where the test start:
```
Comment (tamarasaurus) [inline internal_doc/RUNNING_THE_TESTS.md]: starts

File: internal_doc/RUNNING_THE_TESTS.md [hunk 9]
```diff
@@ -0,0 +1,303 @@
...
+Akeneo PIM uses it for back-end acceptance and end-to-end tests.
+
+It is not possible to run all the tests at once, as there is no `default` profile in the configuration file.
+There are currently 2 profiles: `legacy` (our old deprecated end to end tests), and `acceptance` (for back-end acceptance tests).
+So it is mandatory to specify which profile is to be run.
+
+The configuration is defined in the file [behat.yml.dist](https://github.com/akeneo/pim-community-dev/blob/master/behat.yml.dist).
+You can copy this file as `behat.yml` to change the URLs used in the `legacy` profile, so it fits your Akeneo PIM installation.
+
+If you want to run all the tests of a profile:
```
Comment (rouliane) [inline internal_doc/RUNNING_THE_TESTS.md]: i would add at the end of the sentence (before the ":") ", for example acceptance profile", just to be clear about the parameter "p" in the command line.

File: internal_doc/RUNNING_THE_TESTS.md (hunk 3 above)
Comment (damien-carcel) [reply internal_doc/RUNNING_THE_TESTS.md]: That there will be only `.php_cd` files at the root of each bounded context.

Comment (damien-carcel) [reply internal_doc/RUNNING_THE_TESTS.md]: I change the sentence. We don't really care about the one at the root, as it will disappear.

Comment (tamarasaurus) [reply internal_doc/RUNNING_THE_TESTS.md]: ok !

**Pull request merged**